### DISCUSSION TITLE
Searching for a server for singleplayer mode

### DISCUSSION THREAD

**Discussion by terrarier2111** [Lea-fish/Leafish#117] [General]
We need some (opensource) implementation which we can use for single player mode.

**Comment by PureTryOut** [comment]
Cuberite! Yes it's only 1.12.2 currently and even then missing quite a lot for survival, but it's a compiled language (hello performance!, and no need to install an interpreter like the Java JVM) with an active community and is all around just an awesome project. For a long time I was hoping for a client-side edition of [Cuberite](https://github.com/cuberite/cuberite), and well Leafish could be that.

Having Leafish use it would give it a boost in interest as well, hopefully attracting more developers.

**Reply by PureTryOut** [reply]
I do see a benefit in having a Rust-based server as it would integrate more easily, but the current projects out there (for example [Feather](https://github.com/feather-rs/feather/) and [Quartz](https://github.com/Rusty-Quartz/Quartz) currently have the problem of:

1. being even less complete than Cuberite feature-wise
2. supporting only a single version with seemingly no interest (currently) to support multiple
